File: Builder/java/Builder.java
class Phone{
    String data;
    
    public Phone(){
        data = "";
    }
    
    public String aboutPhone(){
        return data;
    }
    
    public void appendData(String str){
        data += str;
    }
}

interface IDeveloper{
    void createDisplay();
    
    void createBox();
    
    void systemInstall();
    
    Phone getPhone();
}

class AndroidDeveloper implements IDeveloper{
    private Phone phone;
    
    public AndroidDeveloper(){
        phone = new Phone();
    }
    
    @Override
    public void createDisplay(){
        phone.appendData("(create) display Samsung; ");
    }
    
    @Override
    public void createBox(){
        phone.appendData("(create) corpus Samsung; ");
    }
    
    @Override
    public void systemInstall(){
        phone.appendData("(install) system Android; ");
    }
    
    @Override
    public Phone getPhone(){
        return phone;
    }
}

class IphoneDeveloper implements IDeveloper{
    private Phone phone;
    
    public IphoneDeveloper(){
        phone = new Phone();
    }
    
    @Override
    public void createDisplay(){
        phone.appendData("(create) display Iphone; ");
    }
    
    @Override
    public void createBox(){
        phone.appendData("(create) corpus Iphone; ");
    }
    
    @Override
    public void systemInstall(){
        phone.appendData("(install) system IOS; ");
    }
    
    @Override
    public Phone getPhone(){
        return phone;
    }
}

class Director{
    private IDeveloper developer;
    
    public Director(IDeveloper developer){
        this.developer = developer;
    }
    
    public void setDeveloper(IDeveloper developer){
        this.developer = developer;
    }
    
    public Phone MountOnlyPhone(){
        developer.createBox();
        developer.createDisplay();
        return developer.getPhone();
    }
    
    public Phone MountFullPhone(){
        developer.createBox();
        developer.createDisplay();
        developer.systemInstall();
        return developer.getPhone();
    }
}

public class Main {
    public static void main(String[] args){
        IDeveloper developer = new AndroidDeveloper();

        Director director = new Director(developer);

        Phone samsung = director.MountFullPhone();
        System.out.println(samsung.aboutPhone());

        IDeveloper iphoneDeveloper = new IphoneDeveloper();
        director.setDeveloper(iphoneDeveloper);

        Phone iphone = director.MountOnlyPhone();
        System.out.println(iphone.aboutPhone());
    }
}

File: Builder/c#/Builder.cs
class Phone
{
    private string _data;
    
    public Phone() => _data = "";
    
    public string AboutPhone() => _data;
    
    public void AppendData(string str) => _data += str;
}

interface IDeveloper
{
    void CreateDisplay();
    
    void CreateBox();
    
    void SystemInstall();
    
    Phone GetPhone();
}

class AndroidDeveloper : IDeveloper
{
    private Phone _phone;
    
    public AndroidDeveloper() => _phone = new Phone();
    
    public void CreateDisplay() => _phone.AppendData("(create) display Samsung; ");
    
    public void CreateBox() => _phone.AppendData("(create) corpus Samsung; ");
    
    public void SystemInstall() => _phone.AppendData("(install) system Android; ");
    
    public Phone GetPhone() => _phone;
}

class IphoneDeveloper : IDeveloper
{
    private Phone _phone;
    
    public IphoneDeveloper() => _phone = new Phone();
    
    public void CreateDisplay() => _phone.AppendData("(create) display Iphone; ");
    
    public void CreateBox() => _phone.AppendData("(create) corpus Iphone; ");
    
    public void SystemInstall() => _phone.AppendData("(install) system IOS; ");
    
    public Phone GetPhone() => _phone;
}

class Director
{
    private IDeveloper _developer;
    
    public Director(IDeveloper developer) => _developer = developer;
    
    public void SetDeveloper(IDeveloper developer) => _developer = developer;

    public Phone MountOnlyPhone()
    {
        _developer.CreateBox();
        _developer.CreateDisplay();
        return _developer.GetPhone();
    }

    public Phone MountFullPhone()
    {
        _developer.CreateBox();
        _developer.CreateDisplay();
        _developer.SystemInstall();
        return _developer.GetPhone();
    }
}

class Program
{
    static void Main()
    {
        IDeveloper androidDeveloper = new AndroidDeveloper();

        Director director = new Director(androidDeveloper);

        Phone samsung = director.MountFullPhone();
        Console.WriteLine(samsung.AboutPhone());

        IDeveloper iphoneDeveloper = new IphoneDeveloper();
        director.SetDeveloper(iphoneDeveloper);

        Phone iphone = director.MountOnlyPhone();
        Console.WriteLine(iphone.AboutPhone());
    }
}